packages/core/src/shared/utilities/workspaceUtils.ts

```diff
@@ -511,3 +513,87 @@ export function getWorkspaceFoldersByPrefixes(
 
     return results
 }
+
+/**
+ * collects all files that are marked as source
+ * @param sourcePaths the paths where collection starts
+ * @param workspaceFolders the current workspace folders opened
+ * @param respectGitIgnore whether to respect gitignore file
+ * @returns all matched files
+ */
+export async function collectFilesForIndex(
```

Comment:
Why can't this re-use collectFiles? https://github.com/aws/aws-toolkit-vscode/blob/b15135d64f9eec6d9475f133bb48af376801f775/packages/core/src/shared/utilities/workspaceUtils.ts#L304

Could a small adjustment be made to `collectFiles` so that it can be re-used? E.g., add an optional callback parameter that allows the caller to do extra processing on each item?

Code re-use is very, very, very important. It is very important that we avoid copy-pasting code. That has created a lot of tech debt.

Reply:
No for this PR because 
1. collectFiles has hardcoded security scan features like compresses these filles into a zip, it also returns an object with zipFilePath, it also performs file read.  while collect file for index do not do any zip file, do not actually performs the file read operation.
2. collect file for index has hardcoded indexing features like filtering files, ranking them by index priority, etc, which is not there in the collectFiles. And it is not trivial to merge these two.

Reply:
There is no reason that `collectFiles` needs to be coupled with zipping. Please share a backlog item to generalize these "collect" routines and de-duplicate this logic.

Reply:
That `collectFiles` is owned by security scan team. I have created a backlog item.